### ISSUE TITLE
roslyn-ci has test publishing warnings due to too many subresults in test case

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#50337]
I wanted to make note of this warning in the [release branch run](https://dev.azure.com/dnceng/public/_build/results?buildId=938999&view=results) after this PR was merged:

```log
##[warning]No of subresults in test case: Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableContextTests.AnalyzeMethodsInEnabledContextOnly_SpeculativeSemanticModel are more than supported limit 1000, truncating remaning ones.
##[warning]No of subresults in test case: Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableContextTests.AnalyzeMethodsInEnabledContextOnly_01 are more than supported limit 1000, truncating remaning ones.
```

I'm not certain but this indicates a possibility that if these tests start failing in the future that not all the results will be shown in the AzDo UI. Ideally we could make some modification that prevents the warnings from showing in the AzDo run page, either by refactoring the tests or suppressing the warning.

_Originally posted by @RikkiGibson in https://github.com/dotnet/roslyn/issues/49953#issuecomment-754923927_

**Comment by RikkiGibson** [comment]
Tagging @cston
